lec11-modelling.Rmd

```diff
@@ -0,0 +1,527 @@
+use this, we can help walk you through this more.
+
+## Confidence intervals, and p-values
+
+So what's the point of using `glm`? Not only does it create a model with
+constants and coefficients, but it is also used to calculate how *certain* we
+are about the results. The main power of GLM comes from the ability to estimate
+model parameters and derive meaning from how certain $x$ variables (independent
+variables) influence the $y$ (dependent variable). With his model we can use to
+it *predict* what could happen to some value if we knew or could change the
+independent variables ($x$). But we also want to know whether this model
+reflects reality.
+
+From GLM we can determine the *confidence* (or rather, the *uncertainty*) we
+have about the beta ($b$) estimates. In the case of linear regression model, we
+can use `tidy` with the `conf.int` argument to calculate the confidence
+interval.
+
+```{r}
+fit_hf <- glm(weight ~ hindfoot_length, data = portal)
+tidy(fit_hf, conf.int = TRUE) %>% 
+    select(term, estimate, conf.low, conf.high, p.value)
+```
+
+In this case, the uncertainty of the estimate for hindfoot length is between
+2.524 to 2.58, which is pretty narrow (that's a good thing). This tells us that
+the estimate that hindfoot length influences weight is probably reflective of
+reality. That makes sense since longer hindfeet mean there is more space and
+amount of area for more flesh to attach to and that there is more bone, which
+adds weight. We can also be more strict about our uncertainty
+(counterintuitively, a larger confidence interval is more strict).
+
+```{r}
+# Confidence interval of 99%
+tidy(fit_hf, conf.int = TRUE, conf.level = 0.99) %>% 
+    select(term, estimate, conf.low, conf.high, p.value)
+```
+
+We can also expand the model to include more terms.
+
+```{r}
+fit_hfsex <- glm(weight ~ hindfoot_length + sex, data = portal)
+tidy(fit_hfsex, conf.int = TRUE) %>% 
+    select(term, estimate, conf.low, conf.high, p.value)
+```
+
+Great! So in almost all cases, you should favour confidence intervals and
+estimation over the p-value. In some parts of science (especially biomedical
+research), p-values are ubiquitous. But, there is a major problem with p-values.
+What does the p-value *mean*?
+
+P-values are based on and used in hypothesis testing. Hypothesis testing is when
+you ask a question such as "Are males heavier than females in rodents?". There
+is then the null hypothesis ("females and males weigh the same") and the
+alternative hypothesis ("males are heavier than females"). Using statistical
+tests, we can calculate the p-value to give us an indication of which hypothesis
+is true. Traditionally, if the p-value is <0.05 you 'reject' the null
+hypothesis.
+
+However, the true meaning of the p-value is a bit counter intuitive. The
+p-value is the probability of getting a result that is the same as or more
+extreme as what you are getting given the null hypothesis is true. There is
```

Comment:
During the lecture, include an example/rephrasing here, something like "If it is true that females and males weight the same, what is the probability that we would get this (or a more extreme) outcome?"